smart-contracts/contracts/lp-strategy/src/contract.rs

```diff
@@ -501,6 +501,11 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
     // add a new fresh unlocked ibc lock
     IBC_LOCK.save(deps.storage, &Lock::new())?;
 
+    // goodbye join queue
+    while !FAILED_JOIN_QUEUE.is_empty(deps.storage)? {
+        FAILED_JOIN_QUEUE.pop_front(deps.storage)?;
```

Comment:
We shoudl re-add these to trapped errors as a failed join pool so we don't lose them